keccak/Cargo.toml

```diff
@@ -14,5 +14,9 @@ categories = ["cryptography", "no-std"]
 readme = "README.md"
 
 [features]
+asm = []       # Use optimized assembly when available (currently only ARMv8)
 no_unroll = [] # Do no unroll loops for binary size reduction
-simd = [] # Use core::simd (WARNING: requires Nigthly)
+simd = []      # Use core::simd (WARNING: requires Nigthly)
+
+[target.'cfg(target_arch = "aarch64")'.dependencies]
+cpufeatures = "0.2"
```

Comment:
It's a bit unfortunate this is a hard dependency on `aarch64`, although I couldn't figure out how to gate it better

Reply:
What about gating on feature or configuration flag inside the `cfg` statement?

Reply:
This doesn't work:

```toml
[target.'cfg(all(target_arch = "aarch64", feature = "asm"))'.dependencies]
cpufeatures = "0.2"
```

It prints this warning:

> warning: Found `feature = ...` in `target.'cfg(...)'.dependencies`. This key is not supported for selecting dependencies and will not work as expected. Use the [features] section instead: https://doc.rust-lang.org/cargo/reference/features.html

This however, seems to work:

```
[target.'cfg(all(keccak_arvm8, target_arch = "aarch64"))'.dependencies]
cpufeatures = "0.2"
```

So that's a possible argument for using a `cfg` attribute for gating rather than a Cargo feature.